packages/grid/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -104,6 +115,18 @@ export const useGridVirtualScroller = (props: UseGridVirtualScrollerProps) => {
   const [containerWidth, setContainerWidth] = React.useState<number | null>(null);
   const prevTotalWidth = React.useRef(columnsTotalWidth);
 
+  const getNearestIndexToRender = React.useCallback(
```

Comment:
I would be in favor of extracting those methods in an utility file and document them properly (what is offset ? Why we sometime use binary and someting exponential ?)

Reply:
I added some comments explaining. This method is a couple of lines of where it's used. IMO moving it to a separate file will cause more harm than good.